apps/src/templates/teacherDashboard/teacherSectionsRedux.js

```diff
@@ -610,6 +610,25 @@ export function isRosterDialogOpen(state) {
   return getRoot(state).isRosterDialogOpen;
 }
 
+export function oauthProvider(state) {
+  return getRoot(state).provider;
+}
+
+export function sectionCode(state, sectionId) {
+  return (getRoot(state).sections[sectionId] || {}).code;
+}
+
+export function sectionProvider(state, sectionId) {
+  if (isSectionProviderManaged(state, sectionId)) {
+    return oauthProvider(state);
+  }
+  return null;
+}
+
+export function isSectionProviderManaged(state, sectionId) {
+  return !!(getRoot(state).sections[sectionId] || {}).providerManaged;
+}
+
```

Comment:
I think we've talked about this before, but in my selectors, I've usually tried to have the subtree's state passed in instead of global state.

I think this approach is also ok (I like at least that we have the logic for getting to the subtree from the global in a single place), but it might be strange to have these two different approaches. Do you think we should try to be consistent around which approach we take?